05week/checkers.js

```diff
@@ -51,18 +62,56 @@ function Board() {
     }
     console.log(string);
   };
-
-  // Your code here
+  this.populateGrid = () => {
+    // loops through the 8 rows
+    for (let row = 0; row < 8; row++) {
+      // ignores rows which should be empty
+      if (row === 3 || row === 4) continue;
+      // loops through the 8 columns
+      for (let col = 0; col < 8; col++) {
+        // sets current color based on the current row
+        let color = (row < 3 ? 'white' : 'black');
+        // alternates cells to populate with either white or black checkers
+        // then pushes checker to array named checkers
+        if (row % 2 === 0 && col % 2 === 1) {
+          this.grid[row][col] = new Checker(color);
+        } else if (row % 2 === 1 && col % 2 === 0) {
+          this.grid[row][col] = new Checker(color);
+        }
+      }
+    }
+  };
+  
 }
+
 function Game() {
 
   this.board = new Board();
 
   this.start = function() {
     this.board.createGrid();
-    // Your code here
+    this.board.populateGrid();
   };
-}
+
+  /* Need to take start piece and move diagonally left or right.*/
+  this.moveChecker = (rows, columns) => {
+    const start = rows.split('').map(str => Number(str));
```

Comment:
This is unnecessary. You can access letters in a string the exact same way as an array.